Examples/Filtering/OrientImageFilter.cxx

```diff
@@ -0,0 +1,109 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkImage.h"
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+
+#include "itkOrientImageFilter.h"
+
+int
+main(int argc, char * argv[])
+{
+  if (argc < 4)
+  {
+    std::cerr << "Usage: " << std::endl;
+    std::cerr << argv[0]
+              << "  inputImageFile outputImageFile "
+                 "desiredPositiveCoordinateOrientation"
+              << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // Get the inputs from the command line in C++ style
+  std::string inputImageFile = argv[1];
+  std::string outputImageFile = argv[2];
+
+  itk::AnatomicalOrientation desiredCoordinateOrientation{
+    itk::AnatomicalOrientation::CreateFromPositiveStringEncoding(argv[3])
+  };
+
+  if (desiredCoordinateOrientation ==
+      itk::AnatomicalOrientation::PositiveEnum::INVALID)
+  {
+    std::cerr << "Invalid desiredPositiveCoordinateOrientation: " << argv[3]
+              << std::endl;
+    std::cerr << "Valid values are of the form LPS, RIP, etc.";
+    std::cerr << "Where each letter is either L or R, P or A, I or S."
+              << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // Define the image and reader types
+  using PixelType = unsigned short;
+  using ImageType = itk::Image<PixelType, 3>;
+  using ReaderType = itk::ImageFileReader<ImageType>;
+
+  // Create and set up a reader
+  ReaderType::Pointer reader = ReaderType::New();
+  reader->SetFileName(inputImageFile);
+
+  reader->UpdateOutputInformation();
+
+  auto pixelType = reader->GetImageIO()->GetPixelType();
+  auto componentType = reader->GetImageIO()->GetComponentType();
+
+  if (pixelType != itk::IOPixelEnum::SCALAR ||
+      componentType != itk::IOComponentEnum::USHORT)
+  {
+    itkGenericOutputMacro(
+      "The input image is being converted to scalar unsigned short.");
+  }
+
+  auto inputDirection = reader->GetOutput()->GetDirection();
+
+  std::cout << "Input image direction: " << std::endl;
+  inputDirection.PrintSelf(std::cout, itk::Indent(2));
+
+  std::cout << "The input image has the following orientation: " << std::endl;
+  std::cout << '\t' << itk::AnatomicalOrientation(inputDirection)
+            << std::endl;
+
+  std::cout << "The re-orienting to approximately the desired orientation: "
+            << std::endl;
+  std::cout << '\t' << desiredCoordinateOrientation << std::endl;
+
+
+  // Create and set up an OrientImageFilter
+  using OrientImageFilterType = itk::OrientImageFilter<ImageType, ImageType>;
+  OrientImageFilterType::Pointer orienter = OrientImageFilterType::New();
```

Comment:
- Minor suggestion: you may replace `OrientImageFilterType::Pointer` with `auto` here! No need to wait for a follow-up to https://github.com/InsightSoftwareConsortium/ITK/pull/4964#issuecomment-2488812821  😃